### PR TITLE
fix(devcontainer): update devcontainer path and replace unmaintained lsp with maintained one

### PR DESCRIPTION
**Pull request by markkovari**
## Feature or Problem
The devontainer command/tool failed due to unmaintained nix packages and some other configurations.

## Related Issues
#2070 
### Manual Verification

```bash

# go to the root folder
# issue the command

devcontainer up --workspace-folder .
```

#### Additional

I changed the watch in the `.envrc` since it gave me errors/warnings. And little do I know about linkers so if someone has better options for the linker (not gcc) please help.

devcontainer -> `0.65.0`
docker -> `Docker version 26.1.3, build b72abbb`
docker compose -> `Docker Compose version v2.27.3`

### CODE REVIEW

**Review by brooksmtownsend — APPROVED** [review]
Looks good to me, @rvolosatovs anything to change here re: nix?

**Comment by markkovari** [comment]
Thanks @brooksmtownsend 
maybe this addition would help about the lsp @rvolosatovs (which is not searchable and installable anymore) https://github.com/nix-community/rnix-lsp/issues/112 .

Unfortunately the `stdout.cc.cc` did not make the rust-analyzer work with it 😭

**Comment by rvolosatovs** [comment]
Changes look good to me, could you remove the merge commit, however?
I believe just a `git rebase upstream/main` should be enough

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 FROM ghcr.io/xtruder/nix-devcontainer:v1
 
-COPY .devcontainer/etc/nix.conf /etc/nix/nix.conf
+COPY ./etc/nix.conf /etc/nix/nix.conf
```
Comment (rvolosatovs) [inline .devcontainer/Dockerfile:3]: Why the change here?

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -2,8 +2,8 @@ version: '3'
 services:
   dev:
     build:
-      context: ../
-      dockerfile: .devcontainer/Dockerfile
+      context: ../.devcontainer
```
Comment (rvolosatovs) [inline .devcontainer/docker-compose.yml:5]: Correct me if I'm wrong, but I don't think we can do this, since we need access to the `flake.nix` in the root for the nix dev shell to get set up

Comment (markkovari) [reply .devcontainer/docker-compose.yml:5]: Yes, you are correct. I'll rearrange the paths for this.

On the other hand I left a binary reference in the container.json too.
Thanks for the feedback.

Comment (markkovari) [reply .devcontainer/docker-compose.yml:5]: Hey @rvolosatovs I know why, since the dockerignore did not allow to copy the `nix.conf`.

Either we don't ignore the `.devcontainer` folder in the `.dockerignore` and we can copy the `nix.conf`, or we do it somehow manually instead of copying? - Which I think we shouldn't

The issue is that the COPY [here](https://github.com/wasmCloud/wasmCloud/blob/8872acb22a2e0a9b923876e37117cf22973715be/.devcontainer/Dockerfile#L3) cannot access this file at this time - which is the docker build it only exists in the running container after it started and mounted by the docker compose [here](https://github.com/wasmCloud/wasmCloud/blob/8872acb22a2e0a9b923876e37117cf22973715be/.devcontainer/docker-compose.yml#L11-L12).

Can I remove the `.devcontainer` directory on the top level? Is it harmful in any way other than copying it for dev containers?

Comment (rvolosatovs) [reply .devcontainer/docker-compose.yml:5]: Not ignoring `.devcontainer` in the root sounds good to me, the root Dockerfile is only really used for development anyway

File: .envrc [hunk 3]
```diff
@@ -2,5 +2,5 @@ if ! has nix_direnv_version || ! nix_direnv_version 2.3.0; then
   source_url "https://raw.githubusercontent.com/nix-community/nix-direnv/2.3.0/direnvrc" "sha256-Dmd+j63L84wuzgyjITIfSxSD57Tx7v51DMxVZOsiUD8="
 fi
 use flake
-nix_direnv_watch_file rust-toolchain.toml
+watch_file rust-toolchain.toml
```
[the remark this answers is not in the document]
Comment (markkovari) [reply .envrc:5]: Oh, forgot to include. This commad is being deprecated https://github.com/nix-community/nix-direnv/blob/8ab6637a2f7c3240a089f2e0cf5dcefd483252c6/direnvrc#L208

and wanted to eliminate as much warnings as I can.

[the remark this answers is not in the document]
Comment (markkovari) [reply .envrc:5]: And I've seen it like 30 times in the devcontainer logs.

[the remark this answers is not in the document]
Comment (markkovari) [reply .envrc]: reverted this change

**Review by brooksmtownsend — APPROVED** [review]

File: .envrc (hunk 3 above)
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply .envrc]: Oh, I was not aware of that replacing sound good, however, we'll need to update the vendored `nix-direnv` above

**Review by rvolosatovs — APPROVED** [review]
thanks!

File: .envrc (hunk 3 above)
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply .envrc]: Filed https://github.com/wasmCloud/wasmCloud/pull/2441 for this
